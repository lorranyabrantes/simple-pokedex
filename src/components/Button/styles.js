import { StyleSheet } from "react-native";
import { COLORS, SPACING, BORDER_RADIUS } from "../../services/styles";

const styles = StyleSheet.create({
  default: {
    backgroundColor: COLORS.YELLOW,
    height: 35,
    alignItems: "center",
    justifyContent: "center",
    borderRadius: BORDER_RADIUS.medium,
    paddingHorizontal: SPACING.medium,
  },
  disabled: {
    backgroundColor: COLORS.GRAY,
  },
});

export default styles;
