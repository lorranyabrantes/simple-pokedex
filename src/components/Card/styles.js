import { StyleSheet } from "react-native";
import { SPACING, COLORS, BORDER_RADIUS } from "../../services/styles";

const styles = StyleSheet.create({
  container: {
    width: "33.3%",
    padding: SPACING.xsmall,
  },
  content: {
    borderWidth: 1,
    borderRadius: BORDER_RADIUS.medium,
    borderColor: COLORS.GRAY,
    padding: SPACING.small,
    justifyContent: "center",
    alignItems: "center",
  },
  name: {
    marginBottom: SPACING.xsmall,
  },
  image: {
    width: 35,
    height: 35,
  },
});

export default styles;
