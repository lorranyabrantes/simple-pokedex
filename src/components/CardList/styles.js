import { StyleSheet } from "react-native";
import { SPACING } from "../../services/styles";

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    flexWrap: "wrap",
    paddingVertical: SPACING.small,
  },
});

export default styles;
