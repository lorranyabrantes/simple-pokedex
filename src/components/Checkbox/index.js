import React, { useState, useEffect, useCallback } from "react";
import { TouchableOpacity, View, Image } from "react-native";

import iconCheck from "../../assets/png/icon-check.png";

import Text from "../Text";

import styles from "./styles";

const Checkbox = ({ text, onToggle, defaultChecked }) => {
  const [isChecked, setChecked] = useState(defaultChecked);

  const toggleChecked = useCallback(() => {
    setChecked(!isChecked);
    onToggle(!isChecked);
  }, [isChecked, onToggle]);

  useEffect(() => {
    setChecked(defaultChecked);
  }, [defaultChecked]);

  return (
    <View style={styles.container}>
      <TouchableOpacity
        testID="checkbox"
        style={[styles.button, isChecked ? styles.checked : {}]}
        onPress={toggleChecked}
      >
        <Image style={styles.icon} source={iconCheck} />
      </TouchableOpacity>

      {text ? <Text>{text}</Text> : null}
    </View>
  );
};

export default Checkbox;
