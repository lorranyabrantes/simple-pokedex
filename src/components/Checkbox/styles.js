import { StyleSheet } from "react-native";
import { COLORS, SPACING, BORDER_RADIUS } from "../../services/styles";

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
  },
  icon: {
    width: 12,
    height: 12,
  },
  button: {
    width: 20,
    height: 20,
    borderWidth: 2,
    borderColor: COLORS.RED,
    borderRadius: BORDER_RADIUS.small,
    marginRight: SPACING.small,
    alignItems: "center",
    justifyContent: "center",
  },
  checked: {
    backgroundColor: COLORS.RED,
  },
});

export default styles;
