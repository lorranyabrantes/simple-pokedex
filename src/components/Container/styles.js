import { StyleSheet } from "react-native";
import { COLORS, SPACING } from "../../services/styles";

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  scroll: {
    flexGrow: 1,
    backgroundColor: COLORS.WHITE,
    paddingHorizontal: SPACING.small,
  },
});

export default styles;
