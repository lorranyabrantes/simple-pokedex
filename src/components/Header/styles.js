import { StyleSheet, Platform } from "react-native";
import { COLORS, FONT_SIZES, SPACING } from "../../services/styles";

const styles = StyleSheet.create({
  container: {
    backgroundColor: COLORS.RED,
    height: Platform.OS === "ios" ? 80 : 60,
    alignItems: "center",
    justifyContent: "center",
    paddingTop: Platform.OS === "ios" ? 20 : 0,
  },
  title: {
    color: COLORS.WHITE,
    fontSize: FONT_SIZES.xlarge,
  },
  button: {
    position: "absolute",
    top: Platform.OS === "ios" ? 40 : SPACING.large,
    left: SPACING.medium,
  },
  icon: { width: 15, height: 15 },
});

export default styles;
