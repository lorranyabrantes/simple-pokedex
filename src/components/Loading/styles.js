import { StyleSheet } from "react-native";
import { COLORS, SPACING } from "../../services/styles";

const styles = StyleSheet.create({
  container: {
    margin: SPACING.large,
    backgroundColor: COLORS.WHITE,
  },
  loading: {
    flexDirection: "row",
    justifyContent: "center",
    alignContent: "center",
    flex: 1,
  },
});

export default styles;
