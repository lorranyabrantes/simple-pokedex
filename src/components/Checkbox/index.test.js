import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import { COLORS } from "../../services/styles";
import Checkbox from "./";

const props = {
  text: "test",
  defaultChecked: false,
  onToggle: jest.fn(),
};

describe("Checkbox", () => {
  test("should render correctly", async () => {
    const component = render(<Checkbox {...props} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  test("expected the function to be called", async () => {
    const { getByTestId } = render(<Checkbox {...props} />);

    fireEvent.press(getByTestId("checkbox"));

    expect(props.onToggle).toHaveBeenCalled();
  });

  test("expected the checkbox to change color", async () => {
    const { getByTestId } = render(<Checkbox {...props} />);

    fireEvent.press(getByTestId("checkbox"));

    expect(getByTestId("checkbox")).toHaveStyle({
      backgroundColor: COLORS.RED,
    });
  });
});
