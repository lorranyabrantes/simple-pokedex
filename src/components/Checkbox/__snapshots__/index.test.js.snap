// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#EF5350",
        "borderRadius": 3,
        "borderWidth": 2,
        "height": 20,
        "justifyContent": "center",
        "marginRight": 8,
        "opacity": 1,
        "width": 20,
      }
    }
    testID="checkbox"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/png/icon-check.png",
        }
      }
      style={
        Object {
          "height": 12,
          "width": 12,
        }
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#333333",
          "fontFamily": "Roboto-Regular",
          "fontSize": 14,
        },
        undefined,
      ]
    }
  >
    test
  </Text>
</View>
`;
